java-connectors/kafka-connect-gcp-common/src/main/java/io/lenses/streamreactor/connect/gcp/common/auth/GCPConnectionConfig.java

```diff
@@ -15,39 +15,34 @@
  */
 package io.lenses.streamreactor.connect.gcp.common.auth;
 
+import static io.lenses.streamreactor.connect.gcp.common.config.GCPSettings.*;
+
 import io.lenses.streamreactor.common.config.base.RetryConfig;
 import io.lenses.streamreactor.common.config.base.intf.ConnectionConfig;
 import io.lenses.streamreactor.connect.gcp.common.auth.mode.AuthMode;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
 
-import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
-
 @Data
 @Builder
 @AllArgsConstructor
-public class GCPConnectionConfig  implements ConnectionConfig {
+public class GCPConnectionConfig implements ConnectionConfig {
 
-    // TODO: These values are duplicated with GCPConfigSettings.  This will be fixed in the next PR.
-    private static final int HTTP_NUM_OF_RETRIES_DEFAULT = 5;
-    private static final long HTTP_ERROR_RETRY_INTERVAL_DEFAULT = 50L;
+  @Nullable private String projectId;
```

Comment:
I did look into this and made improvements based on this.  Unfortunately you were away on holiday so were unable to give feedback.

It appears that Lombok does not have a way of exposing Optional values in a builder and they [will not add this functionality](https://github.com/projectlombok/lombok/issues/1957) -  In fact this is [against Lombok's ethos](https://github.com/projectlombok/lombok/wiki/Language-Design:-Null-vs.-Optional#lomboks-position-on-null-and-optional).

Based on your Part 1 comments, I did add the default values to the builder so that the defaults will be provided in the case of nulls and this enabled me to remove the code you said [defeated the point of using a builder](https://github.com/lensesio/stream-reactor/pull/1188#discussion_r1585533229).

Reply:
Thank you! We've had additional conversation "behind the scenes" about this, so let's take it one way or the other 👍

Reply:
Please don't be sad 😭 

I killed the JSR annotations.

I am no longer using `Nullable` or `NotNull` but instead using only lombok's `@NotNull`.

Where this code is called is already wrapped in a `Try()` so the exception will be caught and converted to a `ConnectException`.

Where we have `@NotNull` and null is not defined it is a programming error and I want the exception to bubble up to the caller (it should not happen!).  Even the `lombok` builder pattern will support this - if the field is null on `.build()` it will cause the Exception.

Reply:
Thanks, it is better for sure now :) 

But still I think we could either move out of builder to Constructors for nonnull fields or introduce Optionals to getters then handle them nicely whenever one of the fields that we wouldn't like to be null is introduced.

I've sent you patches with two versions as proposition, happy to talk them through if you think one of those could fit the purpose (or if both totally don't ;) )

Reply:
Thanks for the patches, they illustrated your point well - I have added one to the commit.